From ba2a2016a3f3c383b265a99deff8867d0d26ea56 Mon Sep 17 00:00:00 2001
From: "Bryn M. Reeves" <bmr@redhat.com>
Date: Thu, 7 Jul 2016 14:51:16 +0100
Subject: [PATCH] [virtwho] add new plugin for virt-who agent

Add a new plugin for the virtual guest ID reporting agent.

RHBZ#1353552

Signed-off-by: Bryn M. Reeves <bmr@redhat.com>
---
 sos/plugins/virtwho.py | 34 ++++++++++++++++++++++++++++++++++
 1 file changed, 34 insertions(+)
 create mode 100644 sos/plugins/virtwho.py

diff --git a/sos/plugins/virtwho.py b/sos/plugins/virtwho.py
new file mode 100644
index 0000000..e5f3049
--- /dev/null
+++ b/sos/plugins/virtwho.py
@@ -0,0 +1,34 @@
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+
+# You should have received a copy of the GNU General Public License
+# along with this program; if not, write to the Free Software
+# Foundation, Inc., 675 Mass Ave, Cambridge, MA 02139, USA.
+
+from sos.plugins import Plugin, RedHatPlugin
+
+
+class VirtWho(Plugin, RedHatPlugin):
+    """Virt-Who agent
+    """
+
+    plugin_name = 'virtwho'
+    profiles = ('virt', 'system')
+    packages = ('virt-who',)
+
+    def setup(self):
+        self.add_copy_spec("/etc/virt-who.d/*")
+
+    def postproc(self):
+        self.do_path_regex_sub(r"\/etc\/virt-who\.d",
+                               r"(.*password=)(\S*)",
+                               r"\1********")
+
+# vim: set et ts=4 sw=4 :
-- 
2.4.11

